### PR TITLE
[PECO-1803] Splitting the PySql connector into the core and the non core part

### PR DESCRIPTION
**Pull request by jprakash-db**
[PECO-1803]

### Related Links

databricks_sqlalchemy split is present in this PR - https://github.com/databricks/databricks-sqlalchemy/pull/1

## Description
databricks-sql-python library is split so that package size can be reduced for the end user based on their requirements
Particularly pyarrow is the heavy component that is planned to be kept optional 
existing library split into 
 - databricks-sql-connector ( This is kept, so that the existing users import flow does not change )
 - databricks-sql-connector-core ( This is the lightweight library that separates the core part )

## Tasks Completed

- [x] Refractored the code into its respective folders based on the proposed design doc
- [x] pyproject.toml file has been changed to reflect the proper dependencies for the split
- [x] Made sure that all the existing e2e and units tests are working pre and post spit, ensuring parity
- [x] Added benchmarking queries to test the performance of pre and post split and a dashboard has been created for visualization
- [x] Dependency tests are also added to check how the library behaves when certain libraries are not available and the user requests their functions

## How to Test

Testing pipeline remains the same as it is before the split. 
pytest can be used to directly run both the integration as well as unit tests, by `pytest [directory_name or file_name]`

### Addition of dist folder in this repo

Github actions have been setup in the databricks_sqlalchemy repo to run tests using the databricks_sql_connector_core. For running those tests currently we need the .whl file in the dist folder and for temporary testing it has been added to the PR.

Once the library gets published to a public repository such as PyPi then databricks_sqlalchemy will automatically download it from that repo and run the tests using Github actions

## Performance Comparison - Benchmarking

The pre-split and post-split preformance comparison has been made using the large and small queries to make sure their is no regression of performance
Dashboard has been created so that everytime the benchmarking is run the result are stored in the benchfood, and comparisons can be made easily
Screenshot 2024-09-03 at 2 48 19 PM

[PECO-1803]: https://databricks.atlassian.net/browse/PECO-1803?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by madhav-db — APPROVED** [review]

File: databricks_sql_connector_core/src/databricks_sql_connector_core/sql/auth/auth.py [hunk 1]
```diff
@@ -107,9 +107,11 @@ def get_python_sql_connector_auth_provider(hostname: str, **kwargs):
         tls_client_cert_file=kwargs.get("_tls_client_cert_file"),
         oauth_scopes=PYSQL_OAUTH_SCOPES,
         oauth_client_id=kwargs.get("oauth_client_id") or client_id,
-        oauth_redirect_port_range=[kwargs["oauth_redirect_port"]]
-        if kwargs.get("oauth_client_id") and kwargs.get("oauth_redirect_port")
-        else redirect_port_range,
+        oauth_redirect_port_range=(
```
[the remark this answers is not in the document]
Comment (jprakash-db) [reply databricks_sql_connector_core/src/databricks_sql_connector_core/sql/auth/auth.py:110]: @madhav-db  This is the black formatter auto formatting

File: databricks_sql_connector_core/README.md [hunk 2]
```diff
@@ -65,8 +65,8 @@ or to a Databricks Runtime interactive cluster (e.g. /sql/protocolv1/o/123456789
 
 ## Contributing
 
-See [CONTRIBUTING.md](CONTRIBUTING.md)
+See [CONTRIBUTING.md](../CONTRIBUTING.md)
```
[the remark this answers is not in the document]
Comment (jprakash-db) [reply databricks_sql_connector_core/README.md:68]: @madhav-db In the new folder structure Contributing.md file is one level above. So the new link is necessary

File: databricks_sql_connector_core/src/databricks_sql_connector_core/sqlalchemy/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+try:
```
[the remark this answers is not in the document]
Comment (jprakash-db) [reply databricks_sql_connector_core/src/databricks_sql_connector_core/sqlalchemy/__init__.py:1]: @madhav-db In this split databricks_sqlalchemy is kept optional and the requirement is to achieve this without changing the existing import structure for users. That is why this is necessary

File: .gitignore [hunk 4]
```diff
@@ -48,7 +48,7 @@ __pycache__/
 .Python
 build/
 develop-eggs/
-dist/
+#dist/
```
[the remark this answers is not in the document]
Comment (jprakash-db) [reply .gitignore:51]: @kravets-levko I had added the reason for it in the PR description. I have to test the Github actions in the databricks_sqlalchemy - [PR Link](https://github.com/databricks/databricks-sqlalchemy/pull/1) repo, so till the time we don't publish the library to PyPi this was a temporary fix

File: databricks_sql_connector_core/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jackyhu-db) [reply databricks_sql_connector_core/README.md:68]: We should not move the `README.md`, otherwise people will not see anything when they open this git repo

**Review by jackyhu-db — CHANGES_REQUESTED** [review]
We should not move the README.md, otherwise people will not see anything when they open this git repo. Please keep the original README.md and update it with this new package split.

File: databricks_sql_connector_core/src/databricks_sql_connector_core/sql/auth/auth.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jackyhu-db) [reply databricks_sql_connector_core/src/databricks_sql_connector_core/sql/auth/auth.py:110]: please avoid change unnecessary file.

File: databricks_sql_connector_core/src/databricks_sql_connector_core/sql/auth/retry.py [hunk 5]
```diff
@@ -14,7 +14,7 @@
 from urllib3 import Retry
 from urllib3.util.retry import RequestHistory
 
-from databricks.sql.exc import (
+from databricks_sql_connector_core.sql.exc import (
```
Comment (jackyhu-db) [inline databricks_sql_connector_core/src/databricks_sql_connector_core/sql/auth/retry.py:17]: please do not change the namespace and also keep `databricks` namespace as the root.

File: databricks_sql_connector_core/src/databricks_sql_connector_core/sql/client.py [hunk 6]
```diff
@@ -1196,7 +1220,44 @@ def fetchmany_arrow(self, size: int) -> pyarrow.Table:
 
         return results
 
-    def fetchall_arrow(self) -> pyarrow.Table:
+    def merge_columnar(self, result1, result2):
```
Comment (jackyhu-db) [inline databricks_sql_connector_core/src/databricks_sql_connector_core/sql/client.py:1223]: can you move the code change into a separated PR? Let this PR only focus the repo restructure.

File: databricks_sql_connector_core/src/databricks_sql_connector_core/sql/client.py [hunk 7]
```diff
@@ -1196,7 +1220,44 @@ def fetchmany_arrow(self, size: int) -> pyarrow.Table:
...
+        """
+        Function to merge / combining the columnar results into a single result
+
+        :param result1:
+        :param result2:
+        :return:
+        """
+        merged_result = [result1[i] + result2[i] for i in range(len(result1))]
```
Comment (jackyhu-db) [inline databricks_sql_connector_core/src/databricks_sql_connector_core/sql/client.py:1231]: are the len of the results same? Can you add a check before merge?

File: databricks_sql_connector_core/pyproject.toml [hunk 8]
```diff
@@ -14,23 +14,17 @@ thrift = ">=0.16.0,<0.21.0"
...
-sqlalchemy = { version = ">=2.0.21", optional = true }
 openpyxl = "^3.0.10"
+databricks_sqlalchemy = { version = ">=1.0.0", optional = true }
 alembic = { version = "^1.0.11", optional = true }
 urllib3 = ">=1.26"
 
 [tool.poetry.extras]
-sqlalchemy = ["sqlalchemy"]
-alembic = ["sqlalchemy", "alembic"]
+databricks_sqlalchemy = ["databricks_sqlalchemy"]
```
Comment (jackyhu-db) [inline databricks_sql_connector_core/pyproject.toml:26]: why do we need `databricks_sqlalchemy` in `sql_connector_core`, it should NOT have this dependency or declaration.

File: databricks_sql_connector_core/src/databricks_sql_connector_core/sql/__init__.py [hunk 9]
```diff
@@ -1,6 +1,6 @@
 import datetime
 
-from databricks.sql.exc import *
+from databricks_sql_connector_core.sql.exc import *
```
Comment (jackyhu-db) [inline databricks_sql_connector_core/src/databricks_sql_connector_core/sql/__init__.py:3]: we cannot change the package name, otherwise it will break existing consumers.

File: examples/interactive_oauth.py [hunk 10]
```diff
@@ -1,4 +1,4 @@
-from databricks import sql
+from databricks_sql_connector_core import sql
```
Comment (jackyhu-db) [inline examples/interactive_oauth.py:1]: all the examples should still work w/o any code change, we should not change these. If it is required, it means this change will break all the customer codes.

**Comment by kravets-levko** [comment]
As the first iteration, it is more or less okay. Still, there are few major things to fix.

The core package should have most of the dependencies, as well as tests, and so on. The dependencies like `pyarrow` should be optional for it, and thus not installed by default. The core package should keep the original project's structure - in other words, everything inside of `databricks_sql_connector_core` should remain the same as it is now, but just moved into own folder.

The full package should basically have no source files, and the onlt dependency of the core package, but will all its optional dependencies enabled. Please refer to Poetry docs - https://python-poetry.org/docs/pyproject/#extras (look for `extras=["all"]` config of the dependency)

Why it is important and how it should work? When you publish the package, Poetry collects files from the folder where `pyproject.toml` file is located, puts them into archive and uploads to Pypi under the name specified in `pyproject.toml`. Later, when package is installed, `pip` basically just unpacks that archive as is to the folder where all packages are located (`.../lib/python3/site-packages/`). Let's imagine that we created two packages which both contain their sources in the folder `xyz`. After installing both of them, both will be corrupted because their sources will be extracted to the same path on target machine.

So with the configuration described above, we will have all the source code located in the core package. After installation, it will be extracted to the same folders as earlier, so users will not have to update their imports. Optional dependencies will not be installed by default, but if user already has them installed (e.g. `pyarrow`) - correcsponding features will immediately become (remain) available.

Installing the full package will not add any new source files to the user's machine. Everything is already installed with the core package. The full package should just tell `pip` to install optional dependencies.

File: README.md [hunk 11]
```diff
@@ -65,8 +65,8 @@ or to a Databricks Runtime interactive cluster (e.g. /sql/protocolv1/o/123456789
 
 ## Contributing
 
-See [CONTRIBUTING.md](CONTRIBUTING.md)
+See [CONTRIBUTING.md](../CONTRIBUTING.md)
```
Comment (jackyhu-db) [inline README.md:68]: Both `README.md` and `CONTRIBUTING.md` are top level, this change is not required.

File: databricks_sql_connector/README.md
Comment (jackyhu-db) [inline databricks_sql_connector/README.md:1]: Is this "[src/databricks/sql/auth/__init__.py → databricks_sql_connector/README.md]" intended? I guess not and I saw it was empty file.

File: databricks_sql_connector/dist/databricks_sql_connector-1.0.0.tar.gz
Comment (jackyhu-db) [inline databricks_sql_connector/dist/databricks_sql_connector-1.0.0.tar.gz:1]: we should not put any build artifact or `dist` folder in the source repo, please remove it.

File: databricks_sql_connector_core/dist/databricks_sql_connector_core-1.0.0.tar.gz
Comment (jackyhu-db) [inline databricks_sql_connector_core/dist/databricks_sql_connector_core-1.0.0.tar.gz:1]: same as above, please remove them.

File: databricks_sql_connector/pyproject.toml [hunk 12]
```diff
@@ -0,0 +1,33 @@
+[tool.poetry]
+name = "databricks-sql-connector"
+version = "1.0.0"
```
Comment (jackyhu-db) [inline databricks_sql_connector/pyproject.toml:3]: It should be next major version of the existing `databricks-sql-connector`, I believe it is `4.0.0`

File: databricks_sql_connector_core/src/databricks/sql/client.py [hunk 13]
```diff
@@ -1,7 +1,12 @@
 from typing import Dict, Tuple, List, Optional, Any, Union, Sequence
 
 import pandas
-import pyarrow
+
+try:
+    import pyarrow
+except ImportError:
+    pyarrow = None
+
```
Comment (jackyhu-db) [inline databricks_sql_connector_core/src/databricks/sql/client.py:9]: can we move this to the last import

File: databricks_sql_connector_core/src/databricks/sql/thrift_api/TCLIService/TCLIService.py
Comment (jackyhu-db) [inline databricks_sql_connector_core/src/databricks/sql/thrift_api/TCLIService/TCLIService.py:1]: can you avoid changing this file if the change is only for format? Let us try to minimize the change in this PR by avoid unrelated or unnecessary change.

File: databricks_sql_connector_core/src/databricks/sql/auth/auth.py [hunk 14]
```diff
@@ -107,9 +107,11 @@ def get_python_sql_connector_auth_provider(hostname: str, **kwargs):
...
         oauth_scopes=PYSQL_OAUTH_SCOPES,
         oauth_client_id=kwargs.get("oauth_client_id") or client_id,
-        oauth_redirect_port_range=[kwargs["oauth_redirect_port"]]
-        if kwargs.get("oauth_client_id") and kwargs.get("oauth_redirect_port")
-        else redirect_port_range,
+        oauth_redirect_port_range=(
+            [kwargs["oauth_redirect_port"]]
+            if kwargs.get("oauth_client_id") and kwargs.get("oauth_redirect_port")
+            else redirect_port_range
+        ),
```
Comment (jackyhu-db) [inline databricks_sql_connector_core/src/databricks/sql/auth/auth.py:114]: Let us try to minimize the change in this PR by avoid unrelated or unnecessary change (e.g. foramatting).

File: databricks_sql_connector_core/src/databricks/sql/auth/authenticators.py [hunk 15]
```diff
@@ -4,11 +4,17 @@
...
+    infer_cloud_from_host,
+)
 
 # Private API: this is an evolving interface and it will change in the future.
 # Please must not depend on it in your applications.
-from databricks.sql.experimental.oauth_persistence import OAuthToken, OAuthPersistence
+from databricks.sql.experimental.oauth_persistence import (
+    OAuthToken,
+    OAuthPersistence,
+)
```
Comment (jackyhu-db) [inline databricks_sql_connector_core/src/databricks/sql/auth/authenticators.py:17]: same as above. Please avoid format change in this PR.

**Review by jackyhu-db — COMMENTED** [review]
Please remove all the code format change in this PR to minimize the changes in this PR

File: databricks_sql_connector_core/src/databricks/sql/auth/authenticators.py (hunk 15 above)
Comment (jprakash-db) [reply databricks_sql_connector_core/src/databricks/sql/auth/authenticators.py:17]: @jackyhu-db I tried running the black autoformatter. This is the result it is giving, is there some other way to do this ?

File: databricks_sql_connector/pyproject.toml [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+description = "Databricks SQL Connector for Python"
+authors = ["Databricks <databricks-sql-connector-maintainers@databricks.com>"]
+license = "Apache-2.0"
+readme = "README.md"
+include = ["CHANGELOG.md"]
+
+[tool.poetry.dependencies]
+python = "^3.8.0"
+pyarrow = ">=14.0.1,<17"
+databricks_sql_connector_core = { version = ">=1.0.0"}
```
Comment (kravets-levko) [inline databricks_sql_connector/pyproject.toml:13]: This package should basically have no other dependencies except `databricks_sql_connector_core`. Installation of optional deps should be achieved by adding `extras=["all"]` option (like - `databricks_sql_connector_core = { version = ">=1.0.0", extras=["all"]}`. Have you tried it? If there are any caveats with this approach - please share your thoughts

File: databricks_sql_connector/pyproject.toml (hunk 12 above)
Comment (kravets-levko) [reply databricks_sql_connector/pyproject.toml:3]: I'm actually not so sure about it. I think, if the library behavior remains the same for user - no need to change major version number. At least for now we can keep a current version number, and decide how to increment it before the next release

File: databricks_sql_connector_core/src/databricks/sql/auth/authenticators.py (hunk 15 above)
Comment (kravets-levko) [reply databricks_sql_connector_core/src/databricks/sql/auth/authenticators.py:17]: It may be that after your changes some newer version of Black was installed, and it formats files slightly different. You can check this by checking out `main` to a new folder, setting up a devloop, and then comparing to what you have here

File: databricks_sql_connector_core/src/databricks/sql/auth/authenticators.py [hunk 17]
```diff
@@ -25,12 +31,10 @@ class CredentialsProvider(abc.ABC):
     for authenticating requests to Databricks REST APIs"""
 
     @abc.abstractmethod
-    def auth_type(self) -> str:
-        ...
```
Comment (kravets-levko) [inline databricks_sql_connector_core/src/databricks/sql/auth/authenticators.py:29]: Not related to this PR and shouldn't be fixed here, but I've already seen this so much all over our codebase so far. Even though ellipsis literal _can_ be used as a function body, it's not a replacement for `pass`. `pass` actually creates function with no body, while ellipsis literal creates a function with body which consists of that ellipsis literal. It's not any different from using any other literal for this purpose, e.g. `def f(): "hello, world"`. I think after we're done with this splitting, we should go over the code and replace these with `pass`

File: databricks_sql_connector_core/src/databricks/sql/client.py [hunk 18]
```diff
@@ -22,6 +27,7 @@
     ParamEscaper,
     inject_parameters,
     transform_paramstyle,
+    ArrowQueue,
```
Comment (kravets-levko) [inline databricks_sql_connector_core/src/databricks/sql/client.py:30]: Is this import used anywhere in this file? If not - please remove it

File: databricks_sql_connector_core/src/databricks/sql/thrift_api/TCLIService/TCLIService.py
Comment (kravets-levko) [reply databricks_sql_connector_core/src/databricks/sql/thrift_api/TCLIService/TCLIService.py:1]: Even more - this file should not be changed at all because it is generated from `.thrift` definitions (as well as other files in this folder)

File: databricks_sql_connector_core/src/databricks/sql/utils.py [hunk 19]
```diff
@@ -1,5 +1,5 @@
 from __future__ import annotations
-
+import pytz
```
Comment (kravets-levko) [inline databricks_sql_connector_core/src/databricks/sql/utils.py:2]: This import also seems to be a leftover. If not used - please remove it

File: databricks_sql_connector_core/tests/e2e/common/decimal_tests.py [hunk 20]
```diff
@@ -1,14 +1,28 @@
...
+    pyarrow = None
 import pytest
 
 
-class DecimalTestsMixin:
-    decimal_and_expected_results = [
+def decimal_and_expected_results():
+    if pyarrow is None:
+        return []
+
```
Comment (kravets-levko) [inline databricks_sql_connector_core/tests/e2e/common/decimal_tests.py:13]: If the whole test suite is skipped when pyarrow not available - no need to ass additional checks to these functions as they will not be executed anyway (same for other helper functions here)

File: databricks_sql_connector_core/tests/e2e/common/predicates.py [hunk 21]
```diff
@@ -8,9 +8,12 @@
 
 
 def pysql_supports_arrow():
-    """Import databricks.sql and test whether Cursor has fetchall_arrow."""
-    from databricks.sql.client import Cursor
-    return hasattr(Cursor, 'fetchall_arrow')
+    """Checks if the pyarrow library is installed or not"""
+    try:
+        __import__("pyarrow")
```
Comment (kravets-levko) [inline databricks_sql_connector_core/tests/e2e/common/predicates.py:13]: `__import__` is not meant to be used by programmers. Please use regular `import` instead

File: databricks_sql_connector_core/tests/e2e/test_complex_types.py [hunk 22]
```diff
@@ -4,6 +4,9 @@
 from tests.e2e.test_driver import PySQLPytestTestCase
 
 
+@pytest.mark.skipif(
+    pytest.importorskip("pyarrow"), reason="Skipping because pyarrow is not installed"
+)
```
Comment (kravets-levko) [inline databricks_sql_connector_core/tests/e2e/test_complex_types.py:9]: Okay, so I've seen at least three ways to conditionally skip tests when `pyarrow` not available 🙂 Please choose one of them and stick with it everywhere.

Also, please note that here `pytest.importskip` is not used correctly (see https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-importorskip). It should be used in a test body to skip the remaining part of it. `@pytest.mark.skipif` decorator seems to be a better fit if we want to skip the whole test case, but in this case it's better to use it with the predicate you created to check the presense of `pyarrow`

File: poetry.lock
Comment (kravets-levko) [inline poetry.lock:1]: Lockfile should be kept - it's important to have reproducible installs

**Review by kravets-levko — CHANGES_REQUESTED** [review]
Folder structure seems to be good. Two major things to improve - figure out if we can better split dependencies, and also lockfile should be restored/recreated for both new pyprojects (please see my comments)

File: README.md [hunk 23]
```diff
@@ -65,8 +65,8 @@ or to a Databricks Runtime interactive cluster (e.g. /sql/protocolv1/o/123456789
...
 
 ## License
 
-[Apache License 2.0](LICENSE)
+[Apache License 2.0](../LICENSE)
```
[the remark this answers is not in the document]
Comment (jprakash-db) [reply README.md:72]: fixed

File: README.md (hunk 11 above)
Comment (jprakash-db) [reply README.md:68]: fixed

File: databricks_sql_connector/dist/databricks_sql_connector-1.0.0.tar.gz
Comment (jprakash-db) [reply databricks_sql_connector/dist/databricks_sql_connector-1.0.0.tar.gz:1]: fixed

File: databricks_sql_connector_core/dist/databricks_sql_connector_core-1.0.0.tar.gz
Comment (jprakash-db) [reply databricks_sql_connector_core/dist/databricks_sql_connector_core-1.0.0.tar.gz:1]: fixed

File: databricks_sql_connector/README.md
Comment (jprakash-db) [reply databricks_sql_connector/README.md:1]: fixed

File: databricks_sql_connector_core/src/databricks/sql/client.py (hunk 13 above)
Comment (jprakash-db) [reply databricks_sql_connector_core/src/databricks/sql/client.py:9]: fixed

File: databricks_sql_connector_core/src/databricks/sql/client.py (hunk 18 above)
Comment (jprakash-db) [reply databricks_sql_connector_core/src/databricks/sql/client.py:30]: fixed

File: databricks_sql_connector_core/src/databricks/sql/utils.py (hunk 19 above)
Comment (jprakash-db) [reply databricks_sql_connector_core/src/databricks/sql/utils.py:2]: fixed

File: databricks_sql_connector_core/tests/e2e/common/decimal_tests.py (hunk 20 above)
Comment (jprakash-db) [reply databricks_sql_connector_core/tests/e2e/common/decimal_tests.py:13]: fixed

File: databricks_sql_connector_core/tests/e2e/common/predicates.py (hunk 21 above)
Comment (jprakash-db) [reply databricks_sql_connector_core/tests/e2e/common/predicates.py:13]: fixed

File: databricks_sql_connector_core/tests/e2e/test_complex_types.py (hunk 22 above)
Comment (jprakash-db) [reply databricks_sql_connector_core/tests/e2e/test_complex_types.py:9]: fixed

Comment (jprakash-db) [reply databricks_sql_connector_core/tests/e2e/test_complex_types.py:9]: fixed

File: databricks_sql_connector_core/src/databricks/sql/auth/retry.py [hunk 24]
```diff
@@ -77,7 +77,7 @@ class DatabricksRetryPolicy(Retry):
         commands like ExecuteStatement. This is passed to urllib3 by extending its status_forcelist
 
     :param urllib3_kwargs:
-        Dictionary of arguments that are passed to Retry.__init__. Any setting of Retry() that
+        Dictionary of arguments that are passed to Retry.__init__.py. Any setting of Retry() that
```
Comment (kravets-levko) [inline databricks_sql_connector_core/src/databricks/sql/auth/retry.py:80]: These `.py` suffixes (?) should not be here - `Retry.__init__` is a magic method, not a file name

File: databricks_sql_connector_core/src/databricks/sql/client.py [hunk 25]
```diff
@@ -118,7 +125,7 @@ def __init__(
 
                 from databricks.sql.experimental.oauth_persistence import OAuthPersistence, OAuthToken
                 class MyCustomImplementation(OAuthPersistence):
-                    def __init__(self, file_path):
+                    def __init__.py(self, file_path):
```
Comment (kravets-levko) [inline databricks_sql_connector_core/src/databricks/sql/client.py:128]: Same here - `__init__` is a method name, not a file name. Also, how did this code even run? It has to throw a syntax error

File: databricks_sql_connector_core/src/databricks/sql/thrift_backend.py [hunk 26]
```diff
@@ -56,7 +60,7 @@
 TIMESTAMP_AS_STRING_CONFIG = "spark.thriftserver.arrowBasedRowSet.timestampAsString"
 DEFAULT_SOCKET_TIMEOUT = float(900)
 
-# see Connection.__init__ for parameter descriptions.
+# see Connection.__init__.py for parameter descriptions.
```
Comment (kravets-levko) [inline databricks_sql_connector_core/src/databricks/sql/thrift_backend.py:63]: Another one here - `__init__` is a method name, not a file name

File: databricks_sql_connector/pyproject.toml [hunk 27]
```diff
@@ -0,0 +1,23 @@
+[tool.poetry]
+name = "databricks-sql-connector"
+version = "3.3.0"
```
Comment (jackyhu-db) [inline databricks_sql_connector/pyproject.toml:3]: As we change the package structure in this release. I would suggest to bump the major version to `4.0.0` or at least it should be greater than current release `3.4.0`,  so it has to start with `3.5.0`. @kravets-levko wdyt?

**Review by jackyhu-db — APPROVED** [review]

**Review by kravets-levko — APPROVED** [review]
